components/activedirectory/active_directory.go

```diff
@@ -0,0 +1,130 @@
+func (dc *Component) Export(ctx *pulumi.Context, out *Output) error {
+	return components.Export(ctx, dc, out)
+}
+
+// A little structure to help manage state for the Active Directory component
+type activeDirectoryContext struct {
+	pulumiContext    *pulumi.Context
+	comp             *Component
+	timeProvider     *time.Provider
+	createdResources []pulumi.Resource
+}
+
+// NewActiveDirectory creates a new instance of an Active Directory domain deployment
+// Example usage:
+//
+//	activeDirectoryComp, activeDirectoryResources, err := activedirectory.NewActiveDirectory(pulumiContext, awsEnv.CommonEnvironment, host,
+//		activedirectory.WithDomainController("datadogqa.lab", "Test1234#"),
+//	    activedirectory.WithDomainUser("datadogqa.lab\\ddagentuser", "Test5678#"),
+//	)
+//	if err != nil {
+//		return err
+//	}
+//	err = activeDirectoryComp.Export(pulumiContext, &env.ActiveDirectory.Output)
+//	if err != nil {
+//		return err
+//	}
+func NewActiveDirectory(ctx *pulumi.Context, e *config.CommonEnvironment, host *remote.Host, options ...Option) (*Component, []pulumi.Resource, error) {
+	if host.OS.Descriptor().Family() != os.WindowsFamily {
+		// Print flavor in case OS family don't match, as it's more precise than the family (and the .String() conversion already exists).
+		return nil, nil, fmt.Errorf("wrong Operating System family, expected Windows, got %s", host.OS.Descriptor().Flavor.String())
+	}
+
+	params, err := common.ApplyOption(&Configuration{}, options)
+	if err != nil {
+		return nil, nil, err
+	}
+	// Make the resource options that will be passed to the component
+	params.ResourceOptions = append([]pulumi.ResourceOption{pulumi.Parent(host), pulumi.DeletedWith(host)}, params.ResourceOptions...)
+
+	adCtx := activeDirectoryContext{
+		pulumiContext: ctx,
+	}
+
+	domainControllerComp, err := components.NewComponent(*e, host.Name(), func(comp *Component) error {
+		comp.namer = e.CommonNamer.WithPrefix(comp.Name())
+		comp.host = host
+		adCtx.comp = comp
+
+		// We use the time provider multiple times so instantiate it early
+		adCtx.timeProvider, err = time.NewProvider(ctx, comp.namer.ResourceName("time-provider"), &time.ProviderArgs{}, pulumi.DeletedWith(host))
+		if err != nil {
+			return err
+		}
+		adCtx.createdResources = append(adCtx.createdResources, adCtx.timeProvider)
+
+		if params.JoinDomainParams != nil {
+			err = adCtx.joinActiveDirectoryDomain(params.JoinDomainParams)
+			if err != nil {
+				return err
+			}
+		}
+
+		if params.DomainControllerConfiguration != nil {
+			err = adCtx.installDomainController(params.DomainControllerConfiguration)
+			if err != nil {
+				return err
+			}
+		}
+
+		if len(params.DomainUsers) > 0 {
+			// Create users in parallel
+			var createUserResources []pulumi.Resource
+			for _, user := range params.DomainUsers {
+				createDomainUserCmd, err := host.OS.Runner().Command(comp.namer.ResourceName("create-domain-users", user.Username), &command.Args{
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You might want to early check that `Username` do not have duplicates, or this will fail as pulumi expects unique command names